src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -5121,8 +5171,47 @@ RebuildWaitEventSetFlags(WaitEventSet *waitEventSet, List *sessionList)
 			continue;
 		}
 
-		ModifyWaitEvent(waitEventSet, waitEventSetIndex, connection->waitFlags, NULL);
+		bool success =
+			CitusModifyWaitEvent(waitEventSet, waitEventSetIndex,
```

Comment:
would be great to add a test, at least for ModifyWaitEvent

Reply:
Does a failed ModifyWaitEvent have any side effects?

Reply:
It does not seem to have any side effects in the code. And, also when I manually inject errors right after it, all works fine. 

But, to be on the safe side, at the end of `ProcessSessionsWithFailedWaitEventSetOperations`, we might set `execution->rebuildWaitEventSet =true` if any session failed and skip `ProcessWaitEvents` for this loop maybe?

Reply:
(pushed a commit to show, I'll test it a bit further)

Reply:
> (pushed a commit to show, I'll test it a bit further)

I'm testing by adding `ereport(ERROR`, right before/after the functions `AddWaitEventToSet` or `ModifyWaitEvent`. The error is triggered based on the `port`  and `connectionState`.

Sometimes only error for a single node, sometimes multiple, and sometimes all nodes. Also, sometimes error in `MULTI_CONNECTION_CONNECTING` and sometimes `MULTI_CONNECTION_CONNNECTED`.

In all scenarios, when rep > 2, we can fail over to the next node (expect multiple node failures fail the execution). When rep == 1, we fail immediately as expect with a nicer error.